### DISCUSSION TITLE
Custom validation rule support

### DISCUSSION THREAD

**Discussion by rickmills** [statamic/cms#5080] [Q&A]
Hi all,

Not sure if I'm just completely missing this in the docs or if its a missing feature, but is it possible to use custom validation rules within forms in Statamic? 

I know theres the regex rule but thats surprisingly limited in what it can handle, with many advanced regexes just being ignored. 

An example of where this has tripped me up recently would be validating a UK phone number, which is 11 digits, and can have multiple spaces. Its a contact form so you absolutely need to be able to accept a users input having spaces as they're commonly entered that way, but not everyone does it.

Failing this, it would be great if an additional event was registered prior to validation as the `FormSubmitted` event appears to run afterwards.

**Comment by edalzell** [comment]
Do you mean on the front end?

**Reply by rickmills** [reply]
Sorry yes I should've clarified. Specifically on things like contact forms that use a blueprint.

**Comment by ryanmitchell** [comment]
You could hook into the [FormSubmitting](https://statamic.dev/extending/events#formsubmitted) event and add some validation logic there?

**Comment by edalzell** [comment]
Check out this PR: https://github.com/statamic/cms/pull/4949
